### PR TITLE
Add inkchain support

### PR DESCRIPTION
**Pull request by walmat**
Fixes APP-2166

## What changed (plus any additional context for devs)
Adds inkchain support

## Screen recordings / screenshots

https://github.com/user-attachments/assets/1d357d0a-fe4c-4a27-a6cc-fe8641c587d4

## What to test
swaps, user assets, etc.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -309,7 +309,7 @@
     "use-deep-compare": "1.1.0",
     "use-memo-one": "1.1.1",
     "util": "0.10.4",
-    "viem": "2.9.16",
+    "viem": "2.21.54",
```
Comment (walmat) [inline package.json:312]: this bump had a few chains we needed

File: scripts/add_network.sh
Comment (walmat) [inline scripts/add_network.sh:1]: can clean this up or even delete it before merge. Was just easier for me to not have to create the imagesets and other entries by hand

File: src/chains/types.ts [hunk 2]
```diff
@@ -21,7 +28,12 @@ export enum Network {
 }
 
 export enum ChainId {
-  apechain = 33139,
+  test = 123123,
+  zksync = chain.zksync.id,
+  sanko = chain.sanko.id,
+  gravity = chain.gravity.id,
+  ink = -1, // TODO: need Ink chainID
```
Comment (walmat) [inline src/chains/types.ts:35]: note

Comment (walmat) [reply src/chains/types.ts:35]: solved

File: src/languages/en_US.json
[the remark this answers is not in the document]
Comment (walmat) [reply src/languages/en_US.json:1]: Ink resolved

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/network-expanding/ffad4d2610a5d4e0ad1027966aa7ba826b44e864.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/network-expanding/ffad4d2610a5d4e0ad1027966aa7ba826b44e864.ipa) for ffad4d2610a5d4e0ad1027966aa7ba826b44e864

**Review by brunobar79 — APPROVED** [review]
Tested this and added Pow. 
LGTM

**Comment by brunobar79** [comment]
Launch in [simulator](http://localhost:29070/install/ios?virtual=https://app-team.p.rainbow.me/@matthew/network-expanding/ddaf604ab3de20cd5f75945e2dde029983f2d97f.app.zip) or [device](http://localhost:29070/install/ios?physical=https://app-team.p.rainbow.me/@matthew/network-expanding/ddaf604ab3de20cd5f75945e2dde029983f2d97f.ipa) for ddaf604ab3de20cd5f75945e2dde029983f2d97f

**Pull request merged**